### PR TITLE
Build BESS in Ansible script

### PR DESCRIPTION
**Pull request by justinemarie**
The ansible script no longer builds BESS! Fixed -- splits into two files, one for building BESS normally, the other for configuring a vagrant dev environment.

### CODE REVIEW

**Comment by justinemarie** [comment]
Chang, can you double check I didn't screw anything up here? :-)

**Comment by sangjinhan** [comment]
According to Chang, BESS is not built by default on purpose: https://github.com/NetSys/bess/pull/305#issuecomment-280781430

The new script clones a BESS repository _within_ the VM, thus isolated from the host. This is a semantic change from the original script, where the VM "shares" the existing repository in the host via the Vagrant directory sync mechanism (e.g., a user can modify the source code on an OSX host but build and test it on the Linux VM).

**Comment by justinemarie** [comment]
There are now two scripts:
* One for use within vagrant that sets up the development environment --
this is Chang's setup with no changes.
* One for use to set up BESS end to end on a new machine, not inside of
vagrant. This one can be run on bare metal machines or in a VM and does the
entire build process end to end. This is what we recommend people use to
set up BESS on the wiki.

**Comment by justinemarie** [comment]
Re: his comment that some steps take a while, I will add comments to the
ansible files warning users that these steps take a while and it is not
hanging.

File: vagrant/bess.yml [hunk 1]
```diff
@@ -125,12 +11,11 @@
...
 
-    - user: name=vagrant groups=adm,docker append=yes
-      become: true
-
-    - name: Bind /opt/bess as ~/bess
-      mount: name=/home/vagrant/bess fstype=none opts=bind src=/opt/bess state=mounted
-      become: true
+    - name: Clone BESS repository
+      git: repo=https://github.com/NetSys/bess.git dest={{ ansible_env.HOME }}/bess accept_hostkey=yes
+
```
Comment (sangjinhan) [inline vagrant/bess.yml]: There are trailing whitespaces in this line.

File: vagrant/bess.yml [hunk 2]
```diff
@@ -125,12 +11,11 @@
...
-      become: true
-
-    - name: Bind /opt/bess as ~/bess
-      mount: name=/home/vagrant/bess fstype=none opts=bind src=/opt/bess state=mounted
-      become: true
+    - name: Clone BESS repository
+      git: repo=https://github.com/NetSys/bess.git dest={{ ansible_env.HOME }}/bess accept_hostkey=yes
+      
+    - name: Build BESS (this may take a while)
+      shell: ./build.py chdir={{ ansible_env.HOME }}/bess
```
Comment (sangjinhan) [inline vagrant/bess.yml]: One whitespace at the end.

File: vagrant/bess.yml [hunk 3]
```diff
@@ -125,12 +11,11 @@
...
-      mount: name=/home/vagrant/bess fstype=none opts=bind src=/opt/bess state=mounted
-      become: true
+    - name: Clone BESS repository
+      git: repo=https://github.com/NetSys/bess.git dest={{ ansible_env.HOME }}/bess accept_hostkey=yes
+      
+    - name: Build BESS (this may take a while)
+      shell: ./build.py chdir={{ ansible_env.HOME }}/bess 
 
     - name: Set PATH
       lineinfile: dest=/etc/environment line='PATH="/opt/bess/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"'
```
Comment (sangjinhan) [inline vagrant/bess.yml]: I think it should be `{{ ansible_env.HOME }}/bess/bin`, not `/opt/bess/bin`

File: vagrant/bess.yml [hunk 4]
```diff
@@ -1,122 +1,8 @@
...
-    - name: Reboot after 20 sec on kernel panic
-      sysctl: name=kernel.panic value=20 sysctl_set=yes
-      become: true
-
-    - name: Panic on oops
-      sysctl: name=kernel.panic_on_oops value=1 sysctl_set=yes
-      become: true
-
     - name: Enable huge pages
       sysctl: name=vm.nr_hugepages value=1024 sysctl_set=yes
```
Comment (sangjinhan) [inline vagrant/bess.yml]: On a 2-socket NUMA machine, it would fail since each node would allocate 512 2MB pages. But BESS requires 1GB for each socket...

**Comment by sangjinhan** [comment]
tumblr_m4qd7oplis1qd477zo1_1280

**Pull request merged**